pkg/api/config.go

```diff
@@ -103,3 +106,55 @@ func (config *ReleaseBuildConfiguration) DependencyParts(dependency StepDependen
 	}
 	return stream, name, explicit
 }
+
+// WithPresubmitFrom returns a new configuration, where a selected test from the source
+// configuration is injected into the base configuration, together with all elements from
+// the source configuration that are potentially necessary to allow that test to function
+// in the context of the base configuration. The intended use case is to inject the test
+// definition of a "release job" (informing/blocking) into a component ci-operator config
+// to allow dynamically executing any such job on any component PR, without the need to
+// clutter each individual component ci-operator config.
+//
+// WARNING: This code is currently experimental and should not be used outside of the
+// "release jobs on PRs" effort
+// TODO: handle the presubmit/periodic better, extract code etc.
+func (config *ReleaseBuildConfiguration) WithPresubmitFrom(source *ReleaseBuildConfiguration, test string) (*ReleaseBuildConfiguration, error) {
```

Comment:
is `ReleaseBuildConfiguration` user input?  How does this get called?

Reply:
1. ci-operator asks ci-operator-config resolver for component config with a test injected from a different config
2. ci-operator-config resolver returns the result of `WithPresubmitFrom`, where a component config is base and different config is source

`ReleaseBuildConfiguration` is parsed ci-operator config (like https://github.com/openshift/release/blob/master/ci-operator/config/openshift/oc/openshift-oc-master.yaml )

Reply:
Hopefully resolved in Slack convo